x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java

```diff
@@ -426,18 +426,10 @@ public void testNodeShutdownWithUnassignedShards() throws Exception {
         final String nodeBId = getNodeId(nodeB);
         ensureGreen("index");
 
-        updateClusterSettings(Settings.builder().put("cluster.routing.allocation.enable", "none"));
-
-        assertThat(client().admin().indices().prepareFlush("index").get().getSuccessfulShards(), equalTo(2));
-        assertThat(client().admin().indices().prepareRefresh("index").get().getSuccessfulShards(), equalTo(2));
-
-        internalCluster().restartNode(nodeA);
-        internalCluster().restartNode(nodeB);
-
-        assertThat(client().admin().cluster().prepareHealth("index").get().getUnassignedShards(), equalTo(1));
-
         putNodeShutdown(nodeAId, SingleNodeShutdownMetadata.Type.REMOVE, null);
 
+        internalCluster().stopNode(nodeA);
+
         assertBusy(() -> assertNodeShutdownStatus(nodeAId, STALLED));
```

Comment:
The test was getting correct status, however the node was still part of the cluster.
It is updated to stop the node before it is safe according to the api an still getting the same status.

Reply:
I think it is quiet unusual to have unassigned shards in a stable cluster..
Anyways, I randomized the unit test in [TransportGetShutdownStatusActionTests](https://github.com/elastic/elasticsearch/pull/94722/commits/17c1e46cf623787ec72b8027382d2073d029b92b) to cover both scenarios: shard is unassigned and last node may or may not be in cluster.